contrib/node/src/python/pants/contrib/node/tasks/node_build.py

```diff
@@ -58,13 +59,18 @@ def _get_output_dir(self, target, node_installed_path):
       target.payload.output_dir if target.payload.build_script else ''))
 
   def execute(self):
+    targets = self.context.targets(predicate=self.is_node_module)
+    if not targets:
+      return
+
     node_paths = self.context.products.get_data(NodePaths)
-    runtime_classpath_product = self.context.products.get_data(
-      'runtime_classpath', init_func=ClasspathProducts.init_func(self.get_options().pants_workdir))
+    # This is required even if node does not need `compile_classpaths` because some downstream
+    # tasks requires `runtime_classpath` to be initialized correctly with `compile_classpaths`
+    compile_classpath = self.context.products.get_data('compile_classpath')
```

Comment:
What happens if there is no compile_classpath?

Reply:
Made it conditional.